core/src/main/scala/cats/FlatMap.scala

```diff
@@ -66,8 +66,18 @@ import simulacrum.typeclass
    */
   def followedByEval[A, B](fa: F[A])(fb: Eval[F[B]]): F[B] = flatMap(fa)(_ => fb.value)
 
-  /** Sequentially compose two actions, discarding any value produced by the second. */
-  def forEffect[A, B](fa: F[A])(fb: F[B]): F[A] = flatMap(fa)(a => map(fb)(_ => a))
+  /**
+    * Sequentially compose two actions, discarding any value produced by the second.
+    * {{{
+    * scala> import cats._, implicits._
+    * scala> List(1) forEffect Nil
+    * res0: List[Int] = List()
+    * scala> IO('a') forEffect IO(print("b\n")) unsafePerformIO ()
+    * b
+    * res1: Char = 'a'
+    * }}}
+    * */
+  def forEffect[A, B](fa: F[A])(fb: F[B]): F[A] = tapM(fa)(_ => fb)
```

Comment:
There is another PR in flight to move this to a Apply. I don’t think we need tapM here. Can we revert this change?